### PR TITLE
Only show next form with api success #493

### LINKED ISSUE #493: Only show next form with api success

### PR DESCRIPTION
**Pull request by pepermao**
closes #493

### CODE REVIEW

File: src/machine/reviewTaskMachine.ts [hunk 1]
```diff
@@ -87,12 +87,13 @@ export const createNewMachine = ({ value, context }) => {
...
     const t = state.context.utils.t
+    const setCurrentFormAndNextEvents = state.context.utils.setCurrentFormAndNextEvents
     const event = state.event.type
 
     if (event === ReviewTaskEvents.assignUser) {
-        api.createClaimReviewTask({ sentence_hash, machine: state }, t, event)
+        api.createClaimReviewTask({ sentence_hash, machine: state }, t, event, setCurrentFormAndNextEvents)
     } else if (event !== ReviewTaskEvents.init) {
-        api.updateClaimReviewTask({ sentence_hash, machine: state }, t, event)
+        api.updateClaimReviewTask({ sentence_hash, machine: state }, t, event, setCurrentFormAndNextEvents)
```
Comment (thesocialdev) [inline src/machine/reviewTaskMachine.ts:96]: Same as above

File: src/machine/reviewTaskMachine.ts [hunk 2]
```diff
@@ -87,12 +87,13 @@ export const createNewMachine = ({ value, context }) => {
 export const transitionHandler = (state) => {
     const sentence_hash = state.context.reviewData.sentence_hash
     const t = state.context.utils.t
+    const setCurrentFormAndNextEvents = state.context.utils.setCurrentFormAndNextEvents
     const event = state.event.type
 
     if (event === ReviewTaskEvents.assignUser) {
-        api.createClaimReviewTask({ sentence_hash, machine: state }, t, event)
+        api.createClaimReviewTask({ sentence_hash, machine: state }, t, event, setCurrentFormAndNextEvents)
```
[the remark this answers is not in the document]
Comment (pepermao) [reply src/machine/reviewTaskMachine.ts:94]: I just tested and it changes form even giving error
```js
api
    .createClaimReviewTask({ sentence_hash, machine: state }, t, event)
    .then(setCurrentFormAndNextEvents(event))
```

[the remark this answers is not in the document]
Comment (thesocialdev) [reply src/machine/reviewTaskMachine.ts:94]: In that case you need to add a .catch() too

[the remark this answers is not in the document]
Comment (pepermao) [reply src/machine/reviewTaskMachine.ts:94]: I also tried this way, and it does the .then() console

```js
api.createClaimReviewTask({ sentence_hash, machine: state }, t, event)
            .then(value => console.log("then",value))
            .catch(value => console.log("catch", value))
```

[the remark this answers is not in the document]
Comment (pepermao) [reply src/machine/reviewTaskMachine.ts:94]: what is the difference to do in the API call to reviewTaskMachine

**Review by thesocialdev — APPROVED** [review]

File: src/machine/reviewTaskMachine.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (thesocialdev) [reply src/machine/reviewTaskMachine.ts]: The state change should be held in the component as much as possible. Allowing other components and handlers to change the state is a bad pattern, it means the classes are not well designed.

**Pull request merged**